jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/job/Job.java

```diff
@@ -54,12 +61,39 @@
     public String getName();
 
     /**
-     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are allowed.
+     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are
+     * allowed.
      * <p>
      * Default is 2 = medium.
      *
      * @return priority
      */
     public int getPriority();
 
+
+    /**
+     * @return All involved locations
+     */
+    public List<Location> getAllLocations();
```

Comment:
You know the activity, because the Job creates the activities. 
Check Service or Shipment classes!
I don't do anything differently, I only moved the code into the job from the instanceof branches.